### PR TITLE
fix: fix the ippo test and add a test flag

### PR DESCRIPTION
**Pull request by OmaymaMahjoub**
## What?
Add a test flag helping to launch the system in mode TEST_MULTI_THREADING
FIx the ippo test
## Why?
Part of on going efforts to increase testing code coverage of callback redesigned Mava.

### CODE REVIEW

File: mava/systems/jax/launcher.py [hunk 1]
```diff
@@ -43,10 +43,11 @@ def __init__(
         single_process_max_episodes: Optional[int] = None,
         name: str = "System",
         terminal: str = "current_terminal",
+        is_test: Optional[bool] = False,
     ) -> None:
         """Initialise the launcher.
 
-        If multi-process, set up the launchpad program.
+        If multi-process,et up the launchpad program.
```
Comment (RuanJohn) [inline mava/systems/jax/launcher.py]: smakk typo to fix please :)

Comment (OmaymaMahjoub) [reply mava/systems/jax/launcher.py]: yeah I just fixed thanks!

**Review by RuanJohn — CHANGES_REQUESTED** [review]
It looks good @OmaymaMahjoub! One other minor request I have is that we also now only have to copy the node function in the multi process case when we are testing. Could you please add an if statement there as well?

**Comment by OmaymaMahjoub** [comment]
> It looks good @OmaymaMahjoub! One other minor request I have is that we also now only have to copy the node function in the multi process case when we are testing. Could you please add an if statement there as well?

Thanks @RuanJohn  for the review! I updated the PR

**Review by RuanJohn — DISMISSED** [review]
Thank you @OmaymaMahjoub :fire: Ran the test locally with the new changes and it works!

**Comment by OmaymaMahjoub** [comment]
> Thank you @OmaymaMahjoub fire Ran the test locally with the new changes and it works!

Thanks @RuanJohn  let's wait for @DriesSmit  and @AsadJeewa because they came up with an idea to improve the test!

File: mava/systems/jax/launcher.py [hunk 2]
```diff
@@ -166,12 +169,21 @@ def launch(self) -> None:
                 nodes_on_gpu=self._nodes_on_gpu,
             )
 
-            lp.launch(
-                self._program,
-                launch_type=lp.LaunchType.LOCAL_MULTI_PROCESSING,
-                terminal=self._terminal,
-                local_resources=local_resources,
-            )
+            if self._is_test:
```
Comment (DriesSmit) [inline mava/systems/jax/launcher.py:172]: Can we maybe change this to:
```
launch_type = lp.LaunchType.TEST_MULTI_THREADING if self._is_test else lp.LaunchType.LOCAL_MULTI_PROCESSING
```

Comment (DriesSmit) [reply mava/systems/jax/launcher.py:172]: And just asign the launch_type.

**Review by DriesSmit — CHANGES_REQUESTED** [review]
Thanks @OmaymaMahjoub :fire: Just see my one small comment.

File: mava/systems/jax/launcher.py (hunk 2 above)
Comment (OmaymaMahjoub) [reply mava/systems/jax/launcher.py:172]: Okay!

Comment (OmaymaMahjoub) [reply mava/systems/jax/launcher.py]: Done

**Review by DriesSmit — APPROVED** [review]

**Review by AsadJeewa — APPROVED** [review]
Nice!

**Review by RuanJohn — APPROVED** [review]
Thank you @OmaymaMahjoub :muscle:

**Pull request merged**